### ISSUE TITLE
New version numbering will cause issues in Debian

### ISSUE THREAD

**Issue by alexm** [perl5-dbi/DBD-Oracle#109]
I just noticed that the last version released is 1.791. Unfortunately this will cause versioning issues in Debian (and probably other distributions), that do not treat the version number as a whole, i.e. like a decimal. Since 1.791 is considered a greater version than 1.80 in Debian:

```
$ dpkg --compare-versions 1.791 '<=' 1.80 || echo unexpected
unexpected
```

When a new D::O release is made from 1.80 and up, in Debian we'll need to add an epoch to deal with the new numbering, i.e. 1.80 will become `1:1.80-1` instead of `1.80-1`.

I'm not sure that there's any other way to fix it, since I'm guessing that removing 1.791 from CPAN could be painful too, but I thought you may want to know.

**Comment by mjegh** [comment]
I had not idea this was an issue for Debian or any other distribution. My belief was that people in the Perl community had been doing this for years. It is actually not hard to delete 1.791 but it is just as easy to produce 1.80. Hang back and I'll release and bear this in mind for the future.

**Comment by alexm** [comment]
It'd be cool if you released 1.80, but only if you also remove 1.791.
Otherwise, debian/watch will find and still consider 1.791 a greater
version.

In Perl you'll often find versions like 1.002003 to avoid these kind of
versioning issues, but changing the number of decimals is usually
troublesome anyway because not everybody reads version numbers as decimals.


El dj., 25 de jul. 2019, 13:19, Martin J. Evans <notifications@github.com>
va escriure:

**Comment by alexm** [comment]
FWIW, you can change your versioning scheme as long as you change the major
version too (e.g. 2.000000) and it will work everywhere, AFAICT.



El dj., 25 de jul. 2019, 13:19, Martin J. Evans <notifications@github.com>
va escriure:

**Comment by mjegh** [comment]
1.80 released and 1.791 marked for deletion. The deletion takes a few days currently scheduled for 28 Jul 2019 14:26:49 GMT.

**Comment by alexm** [comment]
Thanks for your work, Martin.

Cheers!
Alex

El dj., 25 de jul. 2019, 17:01, Martin J. Evans <notifications@github.com>
va escriure:

**Comment by mjegh** [comment]
Not a problem at all. Whilst I don't use system packages myself for Perl I realise a lot of people do. People like yourself have fed issues/fixes back to me over the years and I've been grateful for them. I was about to release 1.80 instead of 1.791 when it was pointed out to me that 1.791 was possibly a more Perl way of doing it so I did. Had I realised the issue you pointed out I would have just gone with 1.80. No big deal, a few releases over the last few days as much down to my ignorance of distzilla as anything else. Thanks for pointing it out to me.
